README.md

```diff
@@ -155,31 +170,57 @@ val qbeastTable = QbeastTable.forPath(spark, tmpDir)
 
 qbeastTable.getIndexMetrics()
 
-qbeastTable.analyze()
+```
+
+### 6. Optimize the table
+
+**Optimize** is a very expensive operation that consist on **rewriting part of the files to accomplish an even distribution of the records withing files**.
+
+To minimize the write amplification of this command, **we execute it based on subsets of the table**, like `Revision ID's` or specific files.
+
+`Revisions` determine the area of the index. **The last Revision would contain the whole dataset area**, while previous Revisions would contain a _subset_ of the area written before. It can be useful to optimize by specific Revision ID's when we have an incremental appends scenario. 
+
+As an example, imagine you start writting a Dataset with the data from France, and the second time you append records from Germany. The first Revision would contain only the French countries, while the second would enlarge the space to also include Germany. If you are querying only French data, you would be interested in optimizing only the first Revision.
```

Comment:
We need to refine this example, here is some suggested text, but don't take it blindly, please review and refine as needed:

> Say that the initial commit C1 inserting data into the table added data about cities in France, assuming a column called city name that we are indexing with Qbeast. With this commit, we add three files F1, F2 and F3. A second commit C2 adds data about German cities, so it expands the cardinality of the city name column. This commit adds another four files F4, F5, F6 and F7.
>
> The first commit C1 creates a revision R0, while C2 adds another revision R1. Revision R1 is necessary because it expands the cardinality of the city name dimension, and this expansion impacts the index as the range of values changes.
>
> Optimizing consequently can be performed over either revision as we elaborate below. Optimizing, for example, over R0 rewrites files F1, F2 and F3.

Reply:
I would only change that C1 creates revision R1 and C2 creates R2. Revision 0 should be used only for files that are not indexed / staging area. 

Now that I am writing it down, I think we should add that information too.